packages/app/src/hooks/useMetadata.tsx

```diff
@@ -43,6 +43,33 @@ export function useColumns(
   });
 }
 
+export function useJsonColumns(
+  {
+    databaseName,
+    tableName,
+    connectionId,
+  }: {
+    databaseName: string;
+    tableName: string;
+    connectionId: string;
+  },
+  options?: Partial<UseQueryOptions<string[]>>,
+) {
+  return useQuery<string[]>({
+    queryKey: ['useMetadata.useJsonColumns', { databaseName, tableName }],
+    queryFn: async () => {
+      const metadata = getMetadata();
+      return metadata.getJsonColumns({
```

Comment:
style: we can reuse `getColumns` and do the filtering afterward unless `getJsonColumns` is going to be used somewhere else